architecture/2024-05-radius-on-dapr.md

```diff
@@ -0,0 +1,324 @@
+✅ Update local configuration
+```
+
+
+After 
+
+```txt
+✅ Installing Radius 0.32.0 on kind-kind
+✅ Installing Dapr 1.13.0 on kind-kind
+✅ Create new environment default
+   - Kubernetes namespace: default
+   - Recipe pack: local-dev
+✅ Update local configuration
+```
+
+
+**Sample Recipe Contract:**
+
+*n/a*
+
+### Development Experience
+
+*Adding Dapr as a dependency has the potential to impact our development experience. We can follow some existing patterns that Dapr users like to simplify this.*
+
+Wilford is a developer that's making some contributions to Radius for the first time. To get started, he reads the developer document at in [the repo](https://github.com/radius-project/radius/blob/main/docs/contributing/contributing-code/contributing-code-control-plane/running-controlplane-locally.md). He sees that installing Dapr is part of the prerequisites, so he follows the [getting started instructions](https://docs.dapr.io/getting-started/).
+
+After that he follows the rest of the instructions to set up a Kubernetes cluster, and configure his `rad` CLI to talk to the local instance of Radius. When he's done with the setup instructions, he can launch Radius in the debugger. When he does that Radius manages the lifetime of each Dapr process so that Wilford does not to do that manually or learn any extra commands.
+
+## Design
+
+### High Level Design
+
+For our immediate needs we'll be replacing the Radius database and message queue with Dapr's functionality. This doesn't require an architectural change or a large update to the Radius codebase. We'll keep our existing abstractions like the async worker and state store in place. 
+
+### Architecture Diagram
+
+The diagram depicted here shows the organization of resources in Kubernetes. Notably, each of our pods will include Dapr as a sidecar.
+
+We'd use a standard install/configuration of Dapr. The details inside the `dapr-system` are shown for clarity.
+
+Also note that the Radius controller will not need to communicate with Dapr, it communicates only with the UCP API.
+
+![Radius Hosting in Kubernetes](./2024-05-radius-on-dapr/hosting-diagram.png)
+
+<p style="text-align: center;font-style: italic;">
+Fig: Diagram depicting Radius hosting in Kubernetes. <a href="https://excalidraw.com/#json=plLzyHWUCb76t3NnOoiQf,kKABl_YcZOObWznIGKj1jQ">link</a>
+</p>
+
+### Detailed Design
+
+We can break the detailed design into four main categories of work:
+
+- Including Dapr in `rad init` and `rad install`. 
+- Including default component configurations in our Helm Chart.
+- Updating the database implementation.
+- Updating the async work worker.
+- Deleting unused code.
+
+#### Setup
+
+For setup (`rad init`/`rad install`) we previously installed Dapr (a long time ago) along with Contour. We should follow the same design approach we currently do for Contour:
+
+- `rad install` will install Dapr.
+- `rad uninstall` will not uninstall Dapr.
+- `rad install --reinstall` will not upgrade Dapr.
+
```

Comment:
if new version of radius brings in a new version of Dapr, then will there be an upgrade?